tests/core/test_variable.py

```diff
@@ -0,0 +1,349 @@
+        val,state = v.clamp(1.0)
+        self.assertAlmostEqual(val, 1.0)
+        self.assertEqual(state, v.State.FREE)
+        #test above high
+        val,state = v.clamp(2.5)
+        self.assertAlmostEqual(val, 2.0)
+        self.assertEqual(state, v.State.HIGH)
+
+        #construction with no bounds
+        v = relentless.DesignVariable(value=0.0)
+        #test free variable
+        val,state = v.clamp(1.0)
+        self.assertAlmostEqual(val, 1.0)
+        self.assertEqual(state, v.State.FREE)
+
+    def test_value(self):
+        """Test methods for setting values and checking bounds."""
+        #test construction with value between bounds
+        v = relentless.DesignVariable(value=0.0, low=-1.0, high=1.0)
+        self.assertAlmostEqual(v.value, 0.0)
+        self.assertEqual(v.state, v.State.FREE)
+
+        #test below low
+        v.value = -1.5
+        self.assertAlmostEqual(v.value, -1.0)
+        self.assertEqual(v.state, v.State.LOW)
+
+        #test above high
+        v.value = 3
+        self.assertAlmostEqual(v.value, 1.0)
+        self.assertEqual(v.state, v.State.HIGH)
+
+        #test invalid value
+        with self.assertRaises(ValueError):
+            v.value = '0'
+
+    def test_bounds(self):
+        """Test methods for setting bounds and checking value clamping."""
+        #test construction with value between bounds
+        v = relentless.DesignVariable(value=0.0, low=-1.0, high=1.0)
+        self.assertAlmostEqual(v.value, 0.0)
+        self.assertEqual(v.state, v.State.FREE)
+
+        #test changing low bound to above value
+        v.low = 0.2
+        self.assertAlmostEqual(v.value, 0.2)
+        self.assertAlmostEqual(v.low, 0.2)
+        self.assertAlmostEqual(v.high, 1.0)
+        self.assertEqual(v.state, v.State.LOW)
+
+        #test changing high bound to below value
+        v.low = -1.0
+        v.high = -0.2
+        self.assertAlmostEqual(v.value, -0.2)
+        self.assertAlmostEqual(v.low, -1.0)
+        self.assertAlmostEqual(v.high, -0.2)
+        self.assertEqual(v.state, v.State.HIGH)
+
+        #test changing bounds incorrectly
+        with self.assertRaises(ValueError):
+            v.low = 0.2
+        with self.assertRaises(ValueError):
+            v.high = -1.6
+
+class test_SameAs(unittest.TestCase):
+    """Unit tests for relentless.SameAs"""
+
+    def test_init(self):
+        """Test creation with data."""
+        #test variable dependent on DesignVariable
+        v = relentless.DesignVariable(value=1.0)
+        w = relentless.SameAs(v)
+        self.assertEqual(w.value, 1.0)
+        self.assertCountEqual(w.depends, (v,))
+
+        #test variable dependent on DependentVariable
+        u = relentless.SameAs(w)
+        self.assertEqual(u.value, 1.0)
+        self.assertCountEqual(u.depends, (w,))
+
+        #test invalid variable dependence
+        with self.assertRaises(TypeError):
+            u = relentless.SameAs(1.0)
+
+    def test_accessors(self):
```

Comment:
Let's rename all of these `test_value` for each variable class, since that is closer to what you're trying.